webhook/resources/volume/mutator.go

```diff
@@ -97,7 +98,23 @@ func (v *volumeMutator) Create(request *admission.Request, newObj runtime.Object
 	}
 
 	if string(volume.Spec.AccessMode) == "" {
-		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(longhorn.AccessModeReadWriteOnce)))
+		accessModeFromBackup := longhorn.AccessModeReadWriteOnce
+		if volume.Spec.FromBackup != "" {
+			bName, _, _, err := backupstore.DecodeBackupURL(volume.Spec.FromBackup)
+			if err != nil {
+				err := errors.Wrapf(err, "failed to decode backup url %v", volume.Spec.FromBackup)
+				return nil, werror.NewInvalidError(err.Error(), "")
+			}
+			bs, err := v.ds.GetBackupRO(bName)
```

Comment:
```suggestion
			backup, err := v.ds.GetBackupRO(bName)
```

Reply:
I will rename the response variable `bs` to `backup` as suggested for clarity.

https://github.com/longhorn/longhorn-manager/blob/31963edf0b2c3c39aee5840e4e2a0d778cb893f4/webhook/resources/volume/mutator.go#L108-L115